mllib/src/test/scala/org/apache/spark/mllib/fpm/PrefixSpanSuite.scala

```diff
@@ -107,6 +107,75 @@ class PrefixSpanSuite extends SparkFunSuite with MLlibTestSparkContext {
       (Array(4, 5), 2L),
       (Array(5), 3L)
     )
-    assert(compareResult(expectedValue3, result3.collect()))
+    assert(compareResult(expectedValue3.map(x => (x._1.map(Array(_)), x._2)), result3.collect()))
+
+    val sequences4 = Array(
+      "a,abc,ac,d,cf",
+      "ad,c,bc,ae",
+      "ef,ab,df,c,b",
+      "e,g,af,c,b,c")
+    val coder = Array('a', 'b', 'c', 'd', 'e', 'f', 'g').zip(Array(1, 2, 3, 4, 5, 6, 7)).toMap
+    val intSequences = sequences4.map(_.split(",").map(_.toArray.map(coder)))
```

Comment:
Please remove `coder` and test only `PrefixSpan`'s implementation in unit tests. We have a `StringIndexer` to do what `coder` is doing here.